server/context.go

```diff
@@ -123,18 +124,23 @@ type Context struct {
 	// MetricsFrequency determines the frequency at which the server should
 	// record internal metrics.
 	MetricsFrequency time.Duration
+
+	// TimeUntilStoreDead is the time after which if there is no new gossiped
+	// information about a store, it is considered dead.
+	TimeUntilStoreDead time.Duration
 }
 
 // NewContext returns a Context with default values.
 func NewContext() *Context {
 	ctx := &Context{
-		Addr:             defaultAddr,
-		MaxOffset:        defaultMaxOffset,
-		GossipInterval:   defaultGossipInterval,
-		CacheSize:        defaultCacheSize,
-		ScanInterval:     defaultScanInterval,
-		ScanMaxIdleTime:  defaultScanMaxIdleTime,
-		MetricsFrequency: defaultMetricsFrequency,
+		Addr:               defaultAddr,
+		MaxOffset:          defaultMaxOffset,
+		GossipInterval:     defaultGossipInterval,
+		CacheSize:          defaultCacheSize,
+		ScanInterval:       defaultScanInterval,
+		ScanMaxIdleTime:    defaultScanMaxIdleTime,
+		MetricsFrequency:   defaultMetricsFrequency,
+		TimeUntilStoreDead: storage.DefaultTimeUntilStoreDead,
```

Comment:
It can't be in server if I need to use it in testing in storage.  I could make a new const for when we don't want to mark stores as dead if you would prefer that.